app/controllers/diagnostics_controller.rb

```diff
@@ -0,0 +1,79 @@
+    @years_old = params[:years_old].to_i
+    @average = average(@years_old)
+    @result = diagnostics_years_old(@years_old, @user_diagnostic)
+    @items = [ 'rent', 'food_expense', 'utility_bills', 'clothes', 'medical_insurance' , 'communication_cost', 'education_cost', 'entertainment_cost', 'saving', 'anyother' ]
+    @count = count_money_problems(@result)
+  end
+
+  def diagnostic
+    @user_diagnostic = { years_old: params[:years_old], family: params[:family], salary: params[:salary], rent: params[:rent], food_expense: params[:food_expense], utility_bills: params[:utility_bills], clothes: params[:clothes], medical_insurance: params[:medical_insurance], communication_cost: params[:communication_cost], education_cost: params[:education_cost], entertainment_cost: params[:entertainment], saving: params[:saving], anyother: params[:anyother]}
+    redirect_to diagnostics_show_path(result: @user_diagnostic)
+  end
+
+  def diagnostics_years_old(years_old, user_diagnostic)
+    if years_old < 30
+      compare_data(user_diagnostic, MONEY_DATA_20S)
+    elsif years_old < 40
+      compare_data(user_diagnostic, MONEY_DATA_30S)
+    elsif years_old < 50
+      compare_data(user_diagnostic, MONEY_DATA_40S)
+    elsif years_old < 60
+      compare_data(user_diagnostic, MONEY_DATA_50S)
+    elsif years_old < 70
+      compare_data(user_diagnostic, MONEY_DATA_60S)
+    else
+      compare_data(user_diagnostic, MONEY_DATA_OLDERS)
+    end
+  end
+
+  def compare_data(user_diagnostic, money_data)
+    { rent: user_diagnostic[:rent] - money_data[:rent], food_expense: user_diagnostic[:food_expense] - money_data[:food_expense], utility_bills: user_diagnostic[:utility_bills] - money_data[:utility_bills], clothes: user_diagnostic[:clothes] - money_data[:clothes], medical_insurance: user_diagnostic[:medical_insurance] - money_data[:medical_insurance], communication_cost: user_diagnostic[:communication_cost] - money_data[:communication_cost], education_cost: user_diagnostic[:education_cost] - money_data[:education_cost], entertainment_cost: user_diagnostic[:entertainment_cost] - money_data[:entertainment_cost], saving: user_diagnostic[:saving] - money_data[:saving], anyother: user_diagnostic[:anyother] - money_data[:anyother]}
+  end
+
+  def average(years_old)
+    if years_old < 30
+      MONEY_DATA_20S
+    elsif years_old < 40
+      MONEY_DATA_30S
+    elsif years_old < 50
+      MONEY_DATA_40S
+    elsif years_old < 60
+      MONEY_DATA_50S
+    elsif years_old < 70
+      MONEY_DATA_60S
+    else
+      MONEY_DATA_OLDERS
+    end
+  end
+
+  def count_money_problems(result)
+    count = 0
+    result.each do |key, value|
+      if key === :saving && value < 0
```

Comment:
こっちかな

https://techracho.bpsinc.jp/hachi8833/2017_01_19/32941

```suggestion
      if key == :saving && value < 0
```

Reply:
同様な部分を全部==に変更しました